subsys/pm/pm.c

```diff
@@ -148,6 +148,11 @@ bool pm_system_suspend(int32_t kernel_ticks)
 
 	SYS_PORT_TRACING_FUNC_ENTER(pm, system_suspend, kernel_ticks);
 
+	if (!pm_policy_state_any_active()) {
+		/* Return early if all states are unavailable. */
+		return false;
+	}
+
```

Comment:
This _**if**_ breaks the functionality provided by the `pm_state_force` function.
https://docs.zephyrproject.org/latest/doxygen/html/group__subsys__pm__sys.html#ga075be307983f4efdcc93252a31a4258a
```c
/**
 * @brief Force usage of given power state.
 *
 * This function overrides decision made by PM policy forcing
 * usage of given power state upon next entry of the idle thread.
 *
 * @note This function can only run in thread context
 *
 * @param cpu CPU index.
 * @param info Power state which should be used in the ongoing
 *	suspend operation.
 */
bool pm_state_force(uint8_t cpu, const struct pm_state_info *info);
```
My interpretation is that this function should result in overriding restrictions such as pm policy lock or max latency.

Reply:
I've opened #89037 to fix that.